### PR TITLE
fix - max_build_amount return 0 when requirements_met is false

### LINKED ISSUE #125: Change max. build amount to 0 if requirements for ship/defense are not met
Currently the ship / defense popup calculates the max amount of units that can be built with the current resources on the planet.

As seen here on the right side of the screen, the max amount of units that can be built is 12:
Image

TODO:
- The max. build amount should show "0" if the technology requirements for this object are not met (i.e. the object cannot be built). Change the template file for this ajax popup: `/resources/views/ingame/ajax/object.blade.php`.

### PR DESCRIPTION
**Pull request by Danielopes7**
Fixes #125 

I had to reload the player data in `UnitQueueTest.php` because the getResearchLevel in ` PlayerService.php ` was returning 0. This occurred because $this->user_tech hadn't been loaded with the researches yet.

Please let me know if there are any issues 👍

### CODE REVIEW

File: tests/Feature/UnitQueueTest.php [hunk 1]
```diff
@@ -31,6 +32,10 @@ private function basicSetup(): void
         $this->planetSetObjectLevel('research_lab', 1);
         $this->playerSetResearchLevel('energy_technology', 1);
         $this->playerSetResearchLevel('combustion_drive', 1);
+
+        //refresh player
+        $playerService = app()->make(PlayerService::class, ['player_id' => $this->currentUserId]);
+        $playerService->load($this->currentUserId);
```
Comment (lanedirt) [inline tests/Feature/UnitQueueTest.php]: I tried running the tests without this player refresh on my local machine and they still all pass. Could you check again if the tests still run when you remove this code?

**Review by lanedirt — CHANGES_REQUESTED** [review]
Hi @Danielopes7, thanks for the PR. I just tested it on my local machine and it works flawlessly. Good job!

I have one comment regarding the player refresh logic in the test. See below in the review. Could you take a look at this? Thanks for your contribution and help 🚀 !

File: tests/Feature/UnitQueueTest.php (hunk 1 above)
Comment (Danielopes7) [reply tests/Feature/UnitQueueTest.php]: I did the same thing now and it works. I have no idea why it wasn't working before, perhaps the branch was outdate. So I removed this part.

Comment (lanedirt) [reply tests/Feature/UnitQueueTest.php]: 👍 Thanks for double-checking!

**Comment by lanedirt** [comment]
Looking good, merged! Thanks for this PR, much appreciated!

**Pull request merged**